packages/cli/src/controllers/folder.controller.ts

```diff
@@ -67,4 +67,23 @@ export class ProjectController {
 			throw new InternalServerError();
 		}
 	}
+
+	@Delete('/:folderId')
+	@ProjectScope('folder:delete')
+	async deleteFolder(
+		req: AuthenticatedRequest<{ projectId: string; folderId: string }>,
+		_res: Response,
+		@Body payload: DeleteFolderDto,
+	) {
+		const { projectId, folderId } = req.params;
+
+		try {
+			await this.folderService.deleteFolder(folderId, projectId, payload);
+		} catch (e) {
+			if (e instanceof FolderNotFoundError) {
+				throw new NotFoundError(e.message);
+			}
+			throw new InternalServerError();
```

Comment:
Let's not lose the original exception:
```suggestion
			throw new InternalServerError(undefined, e);
```